service/storage_service.cc

```diff
@@ -1526,7 +1526,7 @@ future<> storage_service::await_tablets_rebuilt(raft::server_id replaced_id) {
     };
     if (!is_drained()) {
         slogger.info("Waiting for tablet replicas from the replaced node to be rebuilt");
-        co_await _topology_state_machine.event.wait([&] {
+        co_await _topology_state_machine.event.when([&] {
```

Comment:
Has nothing to do with the bug. Just a cleanup that was sitting in my queue together with the fix. IIRC when() is more efficient with co-routines (otherwise why do we have it).

Reply:
Please add the intent to the commit message.